PubHub/PubHub.Common/Services/UserService.cs

```diff
@@ -0,0 +1,251 @@
+﻿using PubHub.Common.ApiService;
+using PubHub.Common.Models;
+using System.Diagnostics;
+using System.Text;
+using System.Text.Json;
+
+namespace PubHub.Common.Services
+{
+    public class UserService : ServiceRoot
+    {
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly JsonSerializerOptions _serializerOptions;
+
+        internal UserService(IHttpClientFactory clientFactory, string clientName) : base(clientFactory, clientName)
+        {
+            _httpClientFactory = clientFactory;
+
+            _serializerOptions = new JsonSerializerOptions
+            {
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                WriteIndented = true,
+            };
+        }
+
+        // TODO (JBN): Change to GUIDs instead of int when that has been updated.
+
+        /// <summary>
+        /// Calls the API endpoint for adding a <see cref="UserCreateModel"/> to the database.
+        /// </summary>
+        /// <param name="userCreateModel">The <see cref="UserCreateModel"/> holding the new user.</param>
+        /// <returns>A status telling if a user was successfully added to the database.</returns>
+        public async Task<string> AddUser(UserCreateModel userCreateModel)
```

Comment:
A string is kinda odd to work with when trying to figure out if a process succeeded.

Maybe a better solution would be to return a service result of some kind?
This could be as simple as a class looking something like this:
```cs
public class ServiceResult<T>
{
    //    We use an internal constructor because we do not want anyone to set this outside the PubHub.Common library
    internal ServiceResult (int statusCode, T instance, string errorDescriptor)
    {
        //    Set properties
    }
    public int StatusCode { get; }
    public T Instance { get; }
    public string ErrorDescriptor { get; }
}
```

Every service would have to return this result object. This ensures that we always keep it consistent while maintining the integrity of the response.

What do you think, @Sjimooon (_To start conversasion with us all included_) ?

Reply:
Could also just be a boolean but, I do see the safety and better implementation perks of making a service for it as well.

Reply:
I will implement attempt to implement this feature and the other comment you left. 😊